Index: lcd4linux-r1142/event.c
===================================================================
--- lcd4linux-r1142.orig/event.c	2012-01-18 13:18:46.897056929 +0100
+++ lcd4linux-r1142/event.c	2012-01-18 13:20:49.692987389 +0100
@@ -126,7 +126,11 @@
 #if (__GLIBC__ >= 2 && __GLIBC_MINOR__ >= 4)
     int ready = ppoll(fds, j, timeout, NULL);
 #else
-    int ready = poll(fds, j, timeout->tv_sec * 1000 + timeout->tv_nsec / 1000000);
+    int ready;
+    if (timeout->tv_sec * 1000 + timeout->tv_nsec / 1000000 < 0)
+    	ready = poll(fds, j, 0);
+    else
+    	ready = poll(fds, j, timeout->tv_sec * 1000 + timeout->tv_nsec / 1000000);
 #endif
 
     if (ready > 0) {
